core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -218,22 +222,19 @@ static void initLoggerPrefix() {
      * This method is invoked by {@link Elasticsearch#main(String[])}
      * to startup elasticsearch.
      */
-    static void init(String[] args) throws Throwable {
+    static void init(
+            final boolean daemonize,
```

Comment:
I think we should keep this name `foreground`, since we don't actually daemonize from within java and this setting is all about whether we close streams.